server/src/test/java/org/apache/calcite/test/ServerTest.java

```diff
@@ -277,6 +279,156 @@ static Connection connect() throws SQLException {
+          + " i int,\n"
+          + " j int as (i + 1) virtual)\n");
+      s.execute("create table t2 like t including defaults including generated");
+
+      int x = s.executeUpdate("insert into t2 (h, i) values (3, 4)");
+      assertThat(x, is(1));
+
+      final String sql1 = "explain plan for\n"
+          + "insert into t (h, i) values (3, 4)";
+      try (ResultSet r = s.executeQuery(sql1)) {
+        final String sql2 = "explain plan for\n"
+            + "insert into t2 (h, i) values (3, 4)";
+        assertThat(r.next(), is(true));
+        final String plan = r.getString(1);
+        assertThat(r.next(), is(false));
+
+        ResultSet r2 = s.executeQuery(sql2);
+        assertThat(r2.next(), is(true));
+        assertEquals(plan, r2.getString(1).replace("T2", "T"));
+        assertThat(r2.next(), is(false));
+      }
+
+      try (ResultSet r = s.executeQuery("select * from t2")) {
+        assertThat(r.next(), is(true));
+        assertThat(r.getInt("H"), is(3));
+        assertThat(r.wasNull(), is(false));
+        assertThat(r.getInt("I"), is(4));
+        assertThat(r.getInt("J"), is(5)); // j = i + 1
+        assertThat(r.next(), is(false));
+      }
+
+      try {
+        x = s.executeUpdate("insert into t2 values (3, 4, 5)");
+        fail("expected error, got " + x);
+      } catch (SQLException e) {
+        assertThat(e.getMessage(),
+            containsString("Cannot INSERT into generated column 'J'"));
+      }
+    }
+  }
+
+  @Test void testCreateTableLikeWithoutLikeOptions() throws Exception {
+    try (Connection c = connect();
+         Statement s = c.createStatement()) {
+      s.execute("create table t (\n"
+          + " h int not null,\n"
+          + " i int,\n"
+          + " j int as (i + 1) stored,\n"
+          + " k int default -1)");
+      // In table t2, only copy the column and type information from t,
+      // excluding generated expression and default expression
+      s.execute("create table t2 like t");
+
+      int x = s.executeUpdate("insert into t2 (h, i) values (3, 4)");
+      assertThat(x, is(1));
+
+      final String sql1 = "explain plan for\n"
+          + "insert into t2 (h, i) values (3, 4)";
+      try (ResultSet r = s.executeQuery(sql1)) {
+        assertThat(r.next(), is(true));
+        final String plan = ""
+            + "EnumerableTableModify(table=[[T2]], operation=[INSERT], flattened=[false])\n"
+            + "  EnumerableCalc(expr#0..1=[{inputs}], expr#2=[null:INTEGER], proj#0..2=[{exprs}], K=[$t2])\n"
+            + "    EnumerableValues(tuples=[[{ 3, 4 }]])\n";
+        assertThat(r.getString(1), isLinux(plan));
+        assertThat(r.next(), is(false));
+      }
+
+      try (ResultSet r = s.executeQuery("select * from t2")) {
+        assertThat(r.next(), is(true));
+        assertThat(r.getInt("H"), is(3));
+        assertThat(r.wasNull(), is(false));
+        assertThat(r.getInt("I"), is(4));
+        assertThat(r.getInt("J"), is(0)); // excluding generated column
```

Comment:
Why is 0, not null?

Reply:
Because `java.sql.ResultSet#getInt(java.lang.String)` will return 0 if the value is `SQL NULL`. I add `assertThat(r.wasNull(), is(true))` to clarify.